zio-http/src/main/scala/zio/http/Request.scala

```diff
@@ -23,14 +23,29 @@ import zio.ZIO
 import zio.http.internal.HeaderOps
 
 final case class Request(
-  body: Body,
-  headers: Headers,
-  method: Method,
-  url: URL,
-  version: Version,
-  remoteAddress: Option[InetAddress],
+  version: Version = Version.Default,
+  method: Method = Method.Default,
+  url: URL = URL.empty,
+  headers: Headers = Headers.empty,
+  body: Body = Body.empty,
+  remoteAddress: Option[InetAddress] = None,
 ) extends HeaderOps[Request] { self =>
 
+  /**
+   * A right-biased way of combining two requests. Most information will be
+   * merged, but in cases where this does not make sense (e.g. two non-empty
+   * bodies), the information from the right request will be used.
+   */
+  def ++(that: Request): Request =
+    Request(
+      self.version ++ that.version,
+      self.method ++ that.method,
+      self.url ++ that.url,
+      self.headers ++ that.headers,
+      self.body ++ that.body,
+      that.remoteAddress.orElse(self.remoteAddress),
+    )
```

Comment:
I'm not sure to understand what's the semantical meaning nor the purpose of this method.
If I have two Requests, `GET /api/example` and `POST /api/another/example?a=b`, what would that semantically mean to merge these two requests? 
What would be the purpose? And what would be the correct result for this combination?